paasta_tools/autoscaling/cluster_boost.py

```diff
@@ -0,0 +1,160 @@
+"""
+Provides functions to temporary boost cluster capacity.
+Useful during big service bounce or failovers.
+
+This works by setting a temporary multiplier on the cluster's measured load.
+This applies to any resource (cpu, memory, disk)
+If the usage increase overtime, the boost will still apply on top of that.
+Default duration of the boost factor is 40 minutes and default value is 1.5
+"""
+import logging
+from datetime import datetime
+
+from paasta_tools.utils import ZookeeperPool
+
+DEFAULT_BOOST_FACTOR = 1.5
+DEFAULT_BOOST_DURATION = 40
+
+MIN_BOOST_FACTOR = 1.0
+MAX_BOOST_FACTOR = 3.0
+
+MAX_BOOST_DURATION = 240
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class BoostAlreadyActiveError(Exception):
+    pass
+
+
+def get_zk_boost_path(region: str, pool: str) -> str:
+    return '/paasta_cluster_autoscaler/{}/{}/boost'.format(region, pool)
+
+
+def get_boosted_load(region: str, pool: str, current_load: float) -> float:
```

Comment:
I don't see the `zk` injected here, where does it come from?

Reply:
the object zk is initialized inside this function.